app/src/main/java/fr/neamar/kiss/result/AppResult.java

```diff
@@ -47,19 +47,20 @@ public AppResult(AppPojo appPojo) {
     @Override
     public View display(final Context context, int position, View v) {
         if (v == null) {
-            if (appPojo.displayTags.isEmpty()) {
-                v = inflateFromId(context, R.layout.item_app);
-            }
-            else {
-                v = inflateFromId(context, R.layout.item_app_with_tags);
-                TextView tagsView = (TextView) v.findViewById(R.id.item_app_tag);
-                tagsView.setText(enrichText(appPojo.displayTags));
-            }
+            v = inflateFromId(context, R.layout.item_app);
         }
 
         TextView appName = (TextView) v.findViewById(R.id.item_app_name);
         appName.setText(enrichText(appPojo.displayName));
 
+        TextView tagsView = (TextView) v.findViewById(R.id.item_app_tag);
+        if (appPojo.displayTags.isEmpty()) {
+            tagsView.setVisibility(View.GONE);
+        }
+        else {
+            tagsView.setVisibility(View.VISIBLE);
+        }
+        tagsView.setText(enrichText(appPojo.displayTags));
```

Comment:
I was thinking that in the case of recycling the tagsView will contain the old recycled tag string. 
However, it has visibitily.gone so you are probably right